Bmx.CommandLine/CommandLine.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Threading.Tasks;
+using Bmx.Core;
+using Bmx.Idp.Okta;
+
+namespace Bmx.CommandLine {
+	class StubIdp : IIdentityProvider {
+		public string Name => "StubIDP";
+
+		public Task<MfaOption[]> Authenticate( string username, string password ) {
+			throw new NotImplementedException();
+		}
+
+		public Task<bool> ChallengeMfa( int selectedMfaIndex, string challengeResponse ) {
+			throw new NotImplementedException();
+		}
+	}
+
+	public class CommandLine {
+		private BmxCore _bmx;
+		private readonly Parser _cmdLineParser;
+
+		public CommandLine() {
+			// TODO: Fix this :| (ref: Program.cs)
+			_bmx = new BmxCore( new OktaClient( "d2l" ) );
+			_cmdLineParser = BuildCommandLine().UseDefaults().Build();
+
+			_bmx.PromptUserName += GetUser;
+			_bmx.PromptUserPassword += GetPassword;
+			_bmx.PromptMfaType += GetMfaType;
+			_bmx.PromptMfaInput += GetMfaInput;
+			_bmx.PromptRoleSelection += GetRoleType;
```

Comment:
Minor: do we want to go with `m_`, similar to the LMS codebase?

Reply:
AFAIK that naming style is a holdover from older C++ code-bases.

Closest thing to a official .net core [guideline](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md) says prefix "_" for non static private fields. It does also say "m_" if the code-base already uses that style instead :)

Reply:
FYI, there are discrepancies between D2L standard and C# standards out in the world.
https://docs.dev.d2l/index.php/Coding_Standards_-_CSharp#Classes_and_Interfaces